internal/ir/xds.go

```diff
@@ -494,6 +498,12 @@ func (s StringMatch) Validate() error {
 	if s.SafeRegex != nil {
 		matchCount++
 	}
+	if s.Distinct {
```

Comment:
```golang
if s.Name == "" {
			errs = multierror.Append(errs, ErrStringMatchNameIsEmpty)
		}else{
		matchCount++
}
```

Reply:
I dont think `else` is right here, we are appending errors, so both cases (name is empty and match is set) need to be considered